### PR TITLE
feat: make package zero deps

### PR DESCRIPTION
**Pull request by alexkvak**
Minimal supported Node.js version is v16.

Run pipelines on Node.js v16, v18 and v20.

### CODE REVIEW

**Comment by djMax** [comment]
I'd be game for dropping node 14 entirely.

**Review by djMax — APPROVED** [review]

**Comment by djMax** [comment]
Looks good - minor comments if you agree with them, if not also fine. But then we need to get CI passing.

File: src/index.ts [hunk 1]
```diff
@@ -547,8 +542,9 @@ export class Redlock extends EventEmitter {
     // of a failure to extend the lock, and subsequent expiration. In the event
     // of an abort, the error object will be made available at `signal.error`.
     const controller =
-      typeof AbortController === 'undefined'
-        ? new PolyfillAbortController()
+        // AbortController became available as a global in node version 16. Once version 14 reaches its end-of-life, this can be removed.
+        typeof AbortController === 'undefined'
+        ? new (await import('node-abort-controller')).AbortController()
```
Comment (djMax) [inline src/index.ts]: The inline await is a bit "overly punctuated." Maybe a function getAbortController?

Comment (alexkvak) [reply src/index.ts]: removed this code

**Comment by alexkvak** [comment]
Let's try CI now

**Pull request merged**